docs/react/guides/advanced-ssr.md

```diff
@@ -0,0 +1,390 @@
+---
+id: advanced-ssr
+title: Advanced Server Rendering
+---
+
+Welcome to the Advanced Server Rendering guide, where you will learn all about using React Query with streaming, Server Components and the Next.js app router.
+
+You might want to read the [Server Rendering & Hydration guide](../guides/ssr) before this one as it teaches the basics for using React Query with SSR, and [Performance & Request Waterfalls](../guides/request-waterfalls) as well as [Prefetching & Router Integration](../guides/prefetching) also contains valuable background.
+
+We'll be using `useSuspenseQuery` here, so also check out the [Suspense guide](../guides/suspense).
+
+Before we start, let's note that while the `initialData` approach outlined in the SSR guide also works with Server Components, we'll focus this guide on the hydration APIs.
+
+## Server Components & Next.js app router
+
+We won't cover Server Components in depth here, but the short version is that they are components that are guaranteed to always run on the server, not only on the initial page view but **also on page transitions**. This is similar to how Next.js `getServerSideProps`/`getStaticProps` and Remix `loader` works, as these also always run on the server but while those can only return data, Server Components can do a lot more. The data part is central to React Query however, so let's focus on that.
+
+How do we take what we learned in the Server Rendering guide about [passing data prefetched in framework loaders to the app](../guides/ssr#using-the-hydration-apis) and apply that to Server Components and the Next.js app router? The best way to start thinking about this is to consider Server Components as "just" another framework loader.
+
+### A quick note on terminology
+
+So far in these guides, we've been talking about the _server_ and the _client_. It's important to note that confusingly enough this does not match 1-1 with _Server Components_ and _Client Components_. Server Components are guaranteed to always run on the server, but Client Components actually runs in both places. The reason for this being that they are also rendered during the initial _server rendering_ pass.
+
+One way to think of this is that even though Server Components also _render_, they happen during a "loader phase" (always happens on the server), while Client Components run during the "application phase", and that application can run both on the server during SSR, and in for example a browser.
```

Comment:
Is it worth noticing that this is nextJs specific ? Other server component implementations, like [waku](https://github.com/dai-shi/waku), might choose to not render client component on the server at all.

Read that here:
 - https://x.com/dai_shi/status/1668134236589793281
 - https://x.com/dai_shi/status/1668171134691590144

Reply:
Yeah, it's not really Next.js specific, and it currently says "can run", not "does run", but I definitely agree this could be clearer. I'll add a note.